src/Manifest/ManifestManager/Options/OutTableManifestOptions.php

```diff
@@ -0,0 +1,168 @@
+            $result['primary_key'] = $this->primaryKeyColumns;
+        }
+        if (isset($this->delimiter)) {
+            $result['delimiter'] = $this->delimiter;
+        }
+        if (isset($this->enclosure)) {
+            $result['enclosure'] = $this->enclosure;
+        }
+        if (isset($this->columns)) {
+            $result['columns'] = $this->columns;
+        }
+        if (isset($this->incremental)) {
+            $result['incremental'] = $this->incremental;
+        }
+        if (isset($this->metadata)) {
+            $result['metadata'] = $this->metadata;
+        }
+        if (isset($this->columnMetadata)) {
+            $result['column_metadata'] = $this->columnMetadata;
+        }
+        return $result;
+    }
+
+    public function setDestination(string $destination): OutTableManifestOptions
+    {
+        $this->destination = $destination;
+        return $this;
+    }
+
+    /**
+     * @param string[] $primaryKeyColumns
+     * @return OutTableManifestOptions
+     */
+    public function setPrimaryKeyColumns(array $primaryKeyColumns): OutTableManifestOptions
+    {
+        $this->primaryKeyColumns = $primaryKeyColumns;
+        return $this;
+    }
+
+    /**
+     * @param string[] $columns
+     * @return OutTableManifestOptions
+     */
+    public function setColumns(array $columns): OutTableManifestOptions
+    {
+        $this->columns = $columns;
+        return $this;
+    }
+
+    public function setIncremental(bool $incremental): OutTableManifestOptions
+    {
+        $this->incremental = $incremental;
+        return $this;
+    }
+
+    /**
+     * @param mixed[] $metadata
+     * @return OutTableManifestOptions
+     */
+    public function setMetadata(array $metadata): OutTableManifestOptions
+    {
+        foreach ($metadata as $value) {
+            if (!is_array($value)) {
+                throw new OptionsValidationException('Each metadata item must be an array');
+            }
+            $keys = array_keys($value);
+            if ($keys !== ['key', 'value']) {
+                throw new OptionsValidationException('Each metadata item must have only "key" and "value" keys');
+            }
+        }
+        $this->metadata = $metadata;
+        return $this;
+    }
+
+    /**
+     * @param mixed[] $columnsMetadata
+     * @return OutTableManifestOptions
+     */
+    public function setColumnMetadata(array $columnsMetadata): OutTableManifestOptions
+    {
+        foreach ($columnsMetadata as $columnName => $columnMetadata) {
+            if (!is_array($columnMetadata)) {
+                throw new OptionsValidationException('Each column metadata item must be an array');
+            }
+            if (!is_string($columnName)) {
+                throw new OptionsValidationException('Each column metadata item must have string key');
+            }
+            foreach ($columnMetadata as $key => $oneKeyAndValue) {
+                if (!is_array($oneKeyAndValue)) {
```

Comment:
jeste by se to dalo vytahnout do metody ValidateMetadataValue, protoze to je z principu to stejny, akorat normalni metadata jsou pole a column metadata je polepoli